File: readability.py
import re
from math import round

class Readability:
    def __init__(self, l_count=0, w_count=0, s_count=0):
        self.l_count = l_count
        self.w_count = w_count
        self.s_count = s_count

def compute_count(text):
    readable = Readability()
    readable.w_count = 1  # Start with 1 to count the first word

    for char in text:
        # Count letters
        if char.isalpha():
            readable.l_count += 1
        # Count sentences
        if char in ['.', '!', '?']:
            readable.s_count += 1
        # Count words
        if char == ' ':
            readable.w_count += 1

    return readable

def main():
    text = input("Text: ")
    readable = compute_count(text)
    
    # Calculate L and S
    L = round((readable.l_count / readable.w_count) * 100)
    S = round((readable.s_count / readable.w_count) * 100)
    
    # Calculate the Coleman-Liau index
    index = round((0.0588 * L) - (0.296 * S) - 15.8)
    
    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")

if __name__ == "__main__":
    main()

File: week2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <math.h>

// Create readability structure
typedef struct 
{
    float l_count;
    float w_count;
    float s_count;
} Readability;

Readability compute_count(char *text);

int main(void)
{
    char *str = get_string("Text: ");
    Readability readable = compute_count(str);
    float L = round((readable.l_count / readable.w_count) * 100);
    float S = round((readable.s_count / readable.w_count) * 100);
    int index = round((0.0588 * L) - (0.296 * S) - 15.8);
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }

}

Readability compute_count(char *text)
{
    Readability readable;
    readable.l_count = 0;
    readable.w_count = 1;
    readable.s_count = 0;

    int in_word = 0;

    for (int i = 0; text[i] != '\0'; i++)
    {
        char c = text[i];

        // count letters
        if (isalpha(c))
        {
            readable.l_count++;
        }

        // count sentence
        if (c == '.' || c == '!' | c == '?')
        {
            readable.s_count++;
        }

        // count words
        if (c == ' ')
        {
            readable.w_count++;   
        }
    }
    return (readable);
}